### PR TITLE
feat: Added max_tokens and timeout to all LLMs and few embedding adapters

### PR DESCRIPTION
**Pull request by chandrasekharan-zipstack**
## What

- Added `max_tokens` and `timeout` parameters to all supported LLMs and few embedding adapters
- Minor error handling updates for `Mistral`, `Anthropic` and `PaLM`
- Added a default model for `PaLM`
- Moved LLM test connection to `LLMAdapter` base class

## Why

- These parameters need to be user configurable in case of timeout related errors or issues with LLM outputs

## Notes on Testing

- Tried testing the adapter to capture error handling concerns, executed prompts with a few LLMs

## Screenshots

1. Vertex AI
image
1. Azure OpenAI
image
1. Anthropic
image
1. Mistral
image
1. PaLM
image

## Checklist

I have read and understood the [Contribution Guidelines]().

### CODE REVIEW

File: src/unstract/sdk/adapters/embedding/open_ai/src/static/json_schema.json
Comment (gaya3-zipstack) [inline src/unstract/sdk/adapters/embedding/open_ai/src/static/json_schema.json:1]: @chandrasekharan-zipstack Are we changing the title to
"title": "OpenAI Embedding" and similar changes as you pointed out in this PR?

File: src/unstract/sdk/adapters/llm/palm/src/static/json_schema.json [hunk 1]
```diff
@@ -21,15 +21,15 @@
...
+      "default": "models/text-bison-001",
+      "description": "The name of the model to use for this adapter instance. Refer https://ai.google.dev/models/palm. Leave it empty to use the default model."
     },
     "num_output": {
       "type": "number",
       "minimum": 0,
       "multipleOf": 1,
       "title": "Max output tokens",
-      "description": "The number of tokens to generate. The higher the number, the longer the response will be. Leave it empty to use the default value."
+      "description": "The number of tokens to generate. This is limited by the maximum supported by the model and will vary from model to model. The higher the number, the longer the response will be. Leave it empty to use the model's maximum."
```
Comment (gaya3-zipstack) [inline src/unstract/sdk/adapters/llm/palm/src/static/json_schema.json:32]: @chandrasekharan-zipstack How do we guarantee that we set the max to model's maximum when the field is left empty? Do we have some logic to derive the model's max and setting that while passing it on to llama-index?

File: src/unstract/sdk/adapters/embedding/open_ai/src/static/json_schema.json
Comment (chandrasekharan-zipstack) [reply src/unstract/sdk/adapters/embedding/open_ai/src/static/json_schema.json:1]: We could, let me do a find all and replace

File: src/unstract/sdk/adapters/llm/palm/src/static/json_schema.json (hunk 1 above)
Comment (chandrasekharan-zipstack) [reply src/unstract/sdk/adapters/llm/palm/src/static/json_schema.json:32]: @gaya3-zipstack this is set by llama index if we pass the `max_tokens` as `None`. That's why we can't assign a default value here

Comment (gaya3-zipstack) [reply src/unstract/sdk/adapters/llm/palm/src/static/json_schema.json:32]: Ok. Got it. Thank you!

**Review by gaya3-zipstack — APPROVED** [review]
PR looks good.

File: src/unstract/sdk/adapters/llm/palm/src/palm.py [hunk 2]
```diff
@@ -1,12 +1,12 @@
 import os
-from typing import Any
+from typing import Any, Optional
 
+from google.api_core.exceptions import GoogleAPICallError
```
Comment (harini-venkataraman) [inline src/unstract/sdk/adapters/llm/palm/src/palm.py:4]: [MINOR] Are we adding the dependency to the toml, since we refer from library?

**Review by harini-venkataraman — APPROVED** [review]
LGTM

File: src/unstract/sdk/adapters/llm/palm/src/palm.py (hunk 2 above)
Comment (chandrasekharan-zipstack) [reply src/unstract/sdk/adapters/llm/palm/src/palm.py:4]: @harini-venkataraman it comes from our llama index dependencies
image

**Review by Deepak-Kesavan — APPROVED** [review]

**Pull request merged**